SmartDeviceLinkTests/SDLStreamingMediaManagerSpec.m

```diff
@@ -4,13 +4,462 @@
+            });
+
+            it(@"should not attempt to start a new audio and video session", ^{
+                OCMReject([mockAudioLifecycleManager startWithProtocol:[OCMArg any]]);
+                OCMReject([mockVideoLifecycleManager startWithProtocol:[OCMArg any]]);
+            });
+
+            it(@"should notify the delegate object that the secondary transport can be destroyed", ^{
+                OCMVerify([mockSecondaryTransportDelegate destroySecondaryTransport]);
+            });
+        });
+
+        describe(@"switching a service to a different transport", ^{
+            __block SDLProtocol *mockOldProtocol = nil;
+            __block SDLProtocol *mockNewProtocol = nil;
+
+            beforeEach(^{
+                mockOldProtocol = OCMClassMock([SDLProtocol class]);
+                mockNewProtocol = OCMClassMock([SDLProtocol class]);
+
+                OCMStub([mockVideoLifecycleManager endVideoServiceWithCompletionHandler:[OCMArg any]]).andDo(^(NSInvocation *invocation) {
+                    void (^handler)(void);
+                    [invocation getArgument:&handler atIndex:2];
+                    handler();
+                });
+
+                OCMStub([mockAudioLifecycleManager endAudioServiceWithCompletionHandler:[OCMArg any]]).andDo(^(NSInvocation *invocation) {
+                    void (^handler)(void);
+                    [invocation getArgument:&handler atIndex:2];
+                    handler();
+                });
+
+                [testStreamingMediaManager streamingServiceProtocolDidUpdateFromOldVideoProtocol:mockOldProtocol toNewVideoProtocol:mockNewProtocol fromOldAudioProtocol:mockOldProtocol toNewAudioProtocol:mockNewProtocol];
+            });
+
+            it(@"should stop both the audio and video stream managers", ^{
+                OCMVerify([mockAudioLifecycleManager endAudioServiceWithCompletionHandler:[OCMArg any]]);
+                OCMVerify([mockVideoLifecycleManager endVideoServiceWithCompletionHandler:[OCMArg any]]);
+            });
+
+            it(@"should notify the delegate object that the secondary transport can be destroyed", ^{
+                OCMVerify([mockSecondaryTransportDelegate destroySecondaryTransport]);
+            });
+
+            it(@"should tell the audio and video stream managers to destroy the protocol", ^{
+                OCMVerify([mockAudioLifecycleManager destroyProtocol]);
+                OCMVerify([mockVideoLifecycleManager destroyProtocol]);
+            });
+
+            it(@"should try to start a new audio and video session with the new protocol", ^{
+                OCMVerify([mockAudioLifecycleManager startWithProtocol:mockNewProtocol]);
+                OCMVerify([mockVideoLifecycleManager startWithProtocol:mockNewProtocol]);
+
+                expect(testStreamingMediaManager.audioStarted).to(beTrue());
+                expect(testStreamingMediaManager.videoStarted).to(beTrue());
+            });
```

Comment:
These can be combined, maybe the title could just be "should stop both the audio and video stream managers and call the delegate then start a new session"

Reply:
Combined.